task/opm-get-bundle-version/0.1/opm-get-bundle-version.yaml

```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: opm-get-bundle-version
+spec:
+  description: Fetch the current version of the provided OLM bundle image
+  params:
+    - name: bundle-image
+      description: OLM bundle image to query
+  results:
+    - name: bundle-version
+      description: olm.package version
+  steps:
+    - name: opm-get-bundle-version
+      image: "registry.redhat.io/openshift4/ose-operator-registry:latest"
+      env:
+        - name: BUNDLE_IMAGE
+          value: $(params.bundle-image)
+      script: |
+        #!/usr/bin/env bash
+        set -xeo pipefail
+        opm render "${BUNDLE_IMAGE}" \
+          | sed -n '/"type": "olm.package"/,/}/p' \
+          | sed -n 's/.*"version": "\(.*\)".*/\1/p' \
+          | tr -d '\n' \
+          > "$(results.bundle-version.path)"
```

Comment:
That seems rather fragile, e.g. it wouldn't work if the JSON was printed compactly or if the order of the keys was different

```json
        {
            "value": {
                "packageName": "osd-example-operator",
                "version": "4.16.0-28f1940"
            },
            "type": "olm.package"
        }
```

Is there reason to believe that `opm` would always format the output in a way that will work with this sed command?

Reply:
We control the way opm is ran (with json output), and we aren't receiving arbitrarily formatted JSON, I would assume this to be pretty stable. I don't think there is a way to print the json compactly, the only other output option being YAML through the `--output=yaml` flag. I would have preferred to use JQ but I can't find an image with both opm and jq in it. Your suggestion below for using JQ is a great improvement I can follow up with.

The order of the keys shouldn't matter, we search for the `olm.package` object, then the value inside of that.

This also shouldn't really be used outside of our specific use case which is stringing together the custom tasks/pipelines so far which we control the inputs to.

Reply:
> The order of the keys shouldn't matter

Only if `value` came before `type` as in my example. That does seem unlikely though

Reply:
Updated to use your two step recommendation! PTAL when you have a minute, thank you!!